### PR TITLE
Feature/karma

### PR DESCRIPTION
**Pull request by DeepDiver1975**
# Description
Move testing over to karma and test in browser.

# ToDo - maybe followup PRs
- [ ] move remaining tests
- [x] enable files tests
- [ ] add coverage
- [ ] move away from browserification

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        webpack: {},
+
+        webpackMiddleware: {
+            noInfo: true
+        },
+
+        // test results reporter to use
+        // possible values: 'dots', 'progress'
+        // available reporters: https://npmjs.org/browse/keyword/karma-reporter
+        reporters: ['dots'],
```
Comment (PVince81) [inline karma.conf.js]: later on you can enable code coverage output here

File: tests/appsTest.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
+////////////////////////////////
+///////    MAIN TESTS    ///////
+////////////////////////////////
+
+/* globals ownCloud, __karma__ */
+
+describe("Currently testing apps management,", function() {
+// CURRENT TIME
+    var timeRightNow = new Date().getTime();
```
Comment (PVince81) [inline tests/appsTest.js]: if you import the "sinon" JS library, you can also use fake timers to reduce randomness of time-related tests

File: tests/appsTest.js [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        }).catch(error => {
+            expect(error).toBe(null);
+            done();
+        });
+    });
+
+    // there are cors issues with this api
+	describe("App attributes testing,", function() {
+			beforeEach(function(done) {
+				var key = ['attr1', 'attr+plus space', '属性1'];
```
Comment (PVince81) [inline tests/appsTest.js]: weird indent

File: tests/sharingTest.js [hunk 4]
```diff
@@ -0,0 +1,703 @@
+////////////////////////////////
+///////    MAIN TESTS    ///////
+////////////////////////////////
+
+/* globals ownCloud, __karma__ */
+
+// CURRENT TIME
+var timeRightNow = new Date().getTime();
```
Comment (PVince81) [inline tests/sharingTest.js]: I suggest moving all those vars into the first `describe()` block. There's a risk that these will conflict with the values from other files if loaded in sequence.

Comment (DeepDiver1975) [reply tests/sharingTest.js]: in this case I need to duplicate all these vars into two describe blocks - right?

Comment (PVince81) [reply tests/sharingTest.js]: oh, I didn't noticed that. Then wrap the whole file in another `describe()` block then that describes the whole file...

**Comment by DeepDiver1975** [comment]
remaining todos will come in followup prs ->merge

**Pull request merged**